examples/CustomCollectionView/Sample/MosaicCollectionLayoutDelegate.m

```diff
@@ -0,0 +1,168 @@
+//
+//  MosaicCollectionLayoutDelegate.m
+//  Texture
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the /ASDK-Licenses directory of this source tree. An additional
+//  grant of patent rights can be found in the PATENTS file in the same directory.
+//
+//  Modifications to this file made after 4/13/2017 are: Copyright (c) 2017-present,
+//  Pinterest, Inc.  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+
+#import "MosaicCollectionLayoutDelegate.h"
+#import "ImageCellNode.h"
+
+#import <AsyncDisplayKit/ASCollectionElement.h>
+
+@implementation MosaicCollectionLayoutDelegate {
+  // Read-only properties
+  NSInteger _numberOfColumns;
+  CGFloat _headerHeight;
+  CGFloat _columnSpacing;
+  UIEdgeInsets _sectionInset;
+  UIEdgeInsets _interItemSpacing;
+}
+
+- (instancetype)initWithNumberOfColumns:(NSInteger)numberOfColumns headerHeight:(CGFloat)headerHeight
+{
+  self = [super init];
+  if (self != nil) {
+    _numberOfColumns = numberOfColumns;
+    _headerHeight = headerHeight;
+    _columnSpacing = 10.0;
+    _sectionInset = UIEdgeInsetsMake(10.0, 10.0, 10.0, 10.0);
+    _interItemSpacing = UIEdgeInsetsMake(10.0, 0, 10.0, 0);
+  }
+  return self;
+}
+
+- (id)additionalInfoForLayoutWithElements:(ASElementMap *)elements
+{
+  return nil;
+}
+
+- (ASCollectionLayoutState *)calculateLayoutWithContext:(ASCollectionLayoutContext *)context
+{
+  CGFloat layoutWidth = context.viewportSize.width;
+  ASElementMap *elements = context.elements;
+  CGFloat top = 0;
+  
+  NSMapTable<ASCollectionElement *, UICollectionViewLayoutAttributes *> *attrsMap = [NSMapTable mapTableWithKeyOptions:(NSMapTableObjectPointerPersonality | NSMapTableWeakMemory) valueOptions:NSMapTableStrongMemory];
```

Comment:
This is a bit complex for sample code -- add a comment above for why it isn't just a dictionary?